### DISCUSSION TITLE
Wrong region in address tags

### DISCUSSION THREAD

**Discussion by foodev** [osm-search/Nominatim#2649] [Q&A] [answered]
When I search the address `Bischofsweg 20, 01454 Radeberg, GERMANY` I get this result:

![image](https://user-images.githubusercontent.com/2130727/160420877-16e70625-4f30-49f0-9442-1126d479fda3.png)
https://nominatim.openstreetmap.org/search.php?format=json&street=Bischofsweg+20&postalcode=01454&city=Radeberg&countrycodes=de&addressdetails=1

The `address` part contains "Dresdner Heide", which is wrong. Only a small part of "Bischofsweg" is inside the boundary "Dresdner Heide". The housenumber 20 does NOT belong to it.

![image](https://user-images.githubusercontent.com/2130727/160417786-ef93733c-3381-4440-9ca1-387a9a746f1c.png)
https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=191614&class=boundary (the relation at the top is "Dresdner Heide")

In OpenStreetMap, "Dresdner Heide" is part of "Loschwitz", which in turn belongs to "Dresden" (https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=191614&class=boundary). This would mean that the original address is part of "Dresden", which is again wrong.

When I check it via

https://nominatim.openstreetmap.org/details?osmid=185249019&osmtype=W&addressdetails=1&format=json

I get the same result.

I understand that addresses and postcodes are only approximate. Is this the reason for this wrong result? Is there a better way to get the address data or filter the result? Or is this a bug?

**Comment by mtmail** [comment]
It's a common issue when a road crossed boundaries (even country boundaries): the road can only be assigned one parent. There have been some improvements to do an additional check on which side of boundaries a house-number is, especially of a road is also the boundary itself, but I think that only works if there is a second boundary to consider. The village Ullersdorf is only mapped as node.

One data fix would be to map the village boundaries, I assume it's another admin_level=11. Another to split the road at the border to Dresdner Heide.

**Comment by lonvia** [comment, marked as answer]
That's the one special case in the on-the-fly address computation that is not covered yet, see third paragraph in #328.

I wonder if we need to go differently about the problem and explicitly mark boundaries in a street address that only cover part of the street.

**Reply by foodev** [reply]
So is there already an issue for that and you are working on it? Or is there any workaround?

**Reply by lonvia** [reply]
Now there is an issue: #2658
